core/types/block.internal.libevm_test.go

```diff
@@ -0,0 +1,42 @@
+package types
+
+import (
+	"reflect"
+	"slices"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_Body_hasOptionalFields(t *testing.T) {
+	t.Parallel()
+
+	var body Body
+
+	assert.False(t, body.hasOptionalFields(), "body has no optional field set")
+
+	v := reflect.ValueOf(&body).Elem()
+	typ := reflect.TypeOf(body)
```

Comment:
(optional, readability) This is only used once but its placement suggests to the reader that they need to remember it for the entire loop. Consider removing it and replacing its usage with:

```go
fieldType := v.Type().Field(i)
```